final/service-worker.js

```diff
@@ -45,20 +45,50 @@ self.addEventListener('activate', function(e) {
 });
 
 self.addEventListener('fetch', function(e) {
-  console.log('[ServiceWorker] Fetch', e.request.url);
-  var dataUrl = 'https://publicdata-weather.firebaseio.com/';
-  if (e.request.url.indexOf(dataUrl) === 0) {
+  /*
+   * Normalizes a weather data URL like:
+   * https://query.yahooapis.com/v1/public/yql?format=json&q=select * from weather.forecast where woeid=123456
+   * Into:
+   * 123456.json
+   */
+  function normalizeWeatherDataUrl(url) {
+    var substring = url.substring(url.indexOf('woeid'), url.length);
+    var tokens = substring.split('=');
+    var normalizedUrl = tokens[1] + '.json';
+    return normalizedUrl;
+  }
+  console.log('[Service Worker] Fetch', e.request.url);
+  var dataUrl = 'https://query.yahooapis.com/v1/public/yql';
+  if (e.request.url.indexOf(dataUrl) > -1) {
+    /*
+     * When the request URL contains dataUrl, the app is asking for fresh
+     * weather data. In this case, the service worker always goes to the
+     * network and then caches the response. This is called the "Cache then
+     * network" strategy:
+     * https://jakearchibald.com/2014/offline-cookbook/#cache-then-network
+     */
     e.respondWith(
-      fetch(e.request)
-        .then(function(response) {
-          return caches.open(dataCacheName).then(function(cache) {
```

Comment:
Was the `return` on this line a mistake? If you look at Jake's recipe it doesn't do this...